File: src/main/kotlin/net/yttravelers/spotshare/entities/Spot.kt
package net.yttravelers.spotshare.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
class Spot {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var spotId: Int? = null

    @Column(name = "name")
    var spotName: String? = null

    var url: String? = null

    var costExpectation: Int? = null

    var requiredMinutes: Int? = null

    var summary: String? = null

    @ElementCollection(fetch=FetchType.EAGER)
    @CollectionTable(
            name="spotImage"
    )
    @Column(name="imagePath")
    var imagePaths: MutableList<String> = mutableListOf()

    @JsonIgnore
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "spot")
    var routeDetails: MutableList<RouteDetail> = mutableListOf()
}

File: resources/js/src/app/model/spot.ts
import { ICountry } from "./country";

/**
 * スポット
 */
export interface ISpot {
  /** スポットID */
  spotId?: number;
  /**  スポット名称 */
  spotName?: string;
  /** 国 */
  country?: ICountry;
  /** 画像パス */
  imagePaths?: string[];
  /** url */
  url?: string;
  /** メモ */
  memo?: string;
  /** 費用（予算） */
  costExpectation?: number;
  /** 所要時間（時） */
  requiredHours?: number;
  /** 所要時間（分） */
  requiredMinutes?: number;
}